Source/BugsnagNotifier.m

```diff
@@ -359,18 +364,38 @@ - (void)start {
     [self willEnterForeground:self];
 }
 
+- (void)addTerminationObserver:(NSString *)name {
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(stop:)
+                                                 name:name
+                                               object:nil];
+}
+
+/**
+ * Removes observers and listeners to prevent allocations when the app is terminated
+ */
+- (void)stop:(id)sender {
```

Comment:
I'd name this a bit more verbosely, like `unsubscribeFromNotifications`. There is an off chance that with a name like `stop`, it could be picked up erroneously in app review as a private API usage.